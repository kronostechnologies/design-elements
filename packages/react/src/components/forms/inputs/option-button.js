"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importDefault(require("react"));
var styled_components_1 = __importDefault(require("styled-components"));
var Input = styled_components_1.default.input(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  position: absolute;\n  z-index: -1;\n\n  + label {\n    align-items: center;\n    background-color: #fff;\n    border: 1px solid #d9dde2;\n    border-radius: 8px;\n    color: #637282;\n    display: flex;\n    font-size: 24px;\n    justify-content: center;\n    min-height: 40px;\n  }\n\n  &:checked + label {\n    background-color: #006296;\n    border-color: #006296;\n    color: #fff;\n  }\n"], ["\n  position: absolute;\n  z-index: -1;\n\n  + label {\n    align-items: center;\n    background-color: #fff;\n    border: 1px solid #d9dde2;\n    border-radius: 8px;\n    color: #637282;\n    display: flex;\n    font-size: 24px;\n    justify-content: center;\n    min-height: 40px;\n  }\n\n  &:checked + label {\n    background-color: #006296;\n    border-color: #006296;\n    color: #fff;\n  }\n"])));
var OptionButton = function (_a) {
    var checked = _a.checked, label = _a.label, name = _a.name, value = _a.value;
    return (<div>
        <Input checked={checked} id="id" name={name} type="radio" value={value}/>
        <label htmlFor="id">{label}</label>
    </div>);
};
exports.OptionButton = OptionButton;
var templateObject_1;
