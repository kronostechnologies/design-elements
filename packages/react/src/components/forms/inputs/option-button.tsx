import React from 'react';
import styled from 'styled-components';
import uuid from 'uuid/v4';

const Input = styled.input`
  position: absolute;
  z-index: -1;

  + label {
    align-items: center;
    background-color: #fff;
    border: 1px solid #d9dde2;
    border-radius: 8px;
    color: #637282;
    display: flex;
    font-size: 24px;
    justify-content: center;
    min-height: 40px;
  }

  &:checked + label {
    background-color: #006296;
    border-color: #006296;
    color: #fff;
  }
`;

interface OptionButtonProps {
    checked?: boolean;
    label: string;
    name: string;
    value: number;
}

const OptionButton = ({ checked, label, name, value }: OptionButtonProps) => {
    const id = uuid();

    return (
      <div>
          <Input checked={checked} id={id} name={name} type="radio" value={value} />
          <label htmlFor={id}>{label}</label>
      </div>
    );
};

export { OptionButton };
