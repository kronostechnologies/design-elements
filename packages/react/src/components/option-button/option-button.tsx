import { useMemo, VoidFunctionComponent } from 'react';
import styled, { css } from 'styled-components';
import { v4 as uuid } from '../../utils/uuid';
import { useDataAttributes } from '../../hooks/use-data-attributes';

const Input = styled.input`
    ${({ theme }) => css`
            position: absolute;
            z-index: -1;

            + label {
                align-items: center;
                background-color: ${theme.greys.white};
                border: 1px solid ${theme.greys.grey};
                border-radius: var(--border-radius-2x);
                color: ${theme.greys['dark-grey']};
                display: flex;
                font-size: 1.5rem;
                justify-content: center;
                min-height: 40px;
            }

            &:checked + label {
                background-color: ${theme.main['primary-1.1']};
                border-color: ${theme.main['primary-1.1']};
                color: ${theme.greys.white};
            }
      `}
`;

interface OptionButtonProps {
    checked?: boolean;
    className?: string;
    label: string;
    name: string;
    value: number;
}

export const OptionButton: VoidFunctionComponent<OptionButtonProps> = ({
    checked,
    className,
    label,
    name,
    value,
    ...otherProps
}: OptionButtonProps) => {
    const id = useMemo(uuid, []);
    const dataAttributes = useDataAttributes(otherProps);

    return (
        <div className={className}>
            <Input
                checked={checked}
                id={id}
                name={name}
                type="radio"
                value={value}
                {...dataAttributes /* eslint-disable-line react/jsx-props-no-spreading */}
            />
            <label htmlFor={id}>{label}</label>
        </div>
    );
};
