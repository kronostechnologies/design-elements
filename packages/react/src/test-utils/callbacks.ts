export function doNothing(): void {
    // do nothing
}
