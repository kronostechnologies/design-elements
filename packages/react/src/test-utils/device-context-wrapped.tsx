import React, { ReactElement } from 'react';
import { DeviceContextProvider, DeviceType } from '../components/device-context-provider/device-context-provider';

export function DeviceContextWrapped(children: ReactElement, device?: DeviceType): ReactElement {
    return (
        <DeviceContextProvider staticDevice={device}>
            {children}
        </DeviceContextProvider>
    );
}
