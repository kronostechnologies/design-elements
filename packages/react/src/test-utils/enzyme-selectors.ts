import { ReactWrapper, ShallowWrapper } from 'enzyme';

enum ModifierEnum {
    StartsWith = '^',
    EndsWith = '$',
    ContainsOptionalHyphen = '|',
    Contains = '*',
    ContainsWord = '~',
    None = '',
}
type SelectorModifier = `${ModifierEnum}`;

function filterHtmlNodesOnly<W extends ReactWrapper | ShallowWrapper>(
    nodes: W,
): W {
    return nodes
        .hostNodes()
        .filterWhere((x) => x.name() !== null && !x.name().startsWith('styled.')) as W;
}

export function findByTestId<W extends ReactWrapper | ShallowWrapper>(
    wrapper: W,
    testId: string,
    options?: {
        modifier?: SelectorModifier,
        nodeName?: string,
        // Don't use with shallow wrapper as it contains styled node over the html nodes.
        htmlNodesOnly?: boolean,
    },
): W {
    const modifier = options?.modifier ?? '';

    let nodes = wrapper.find(`[data-testid${modifier}="${testId}"]`) as W;

    if (options?.nodeName) {
        nodes = nodes.filterWhere((x) => x.name() === options.nodeName) as W;
    }
    if (options?.htmlNodesOnly) {
        nodes = filterHtmlNodesOnly(nodes);
    }

    return nodes;
}

export function getByTestId<W extends ReactWrapper | ShallowWrapper>(
    wrapper: W,
    testId: string,
    options?: {
        modifier?: SelectorModifier,
        nodeName?: string,
        // Don't use with shallow wrapper as it contains styled node instead of the html nodes.
        htmlNodesOnly?: boolean,
    },
): W {
    return findByTestId(wrapper, testId, options).at(0) as W;
}
