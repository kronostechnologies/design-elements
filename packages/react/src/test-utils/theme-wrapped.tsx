import React, { ComponentProps, ComponentType, ReactElement } from 'react';
import { equisoftTheme, ThemeWrapper } from '..';

type ThemeProps = Pick<ComponentProps<typeof ThemeWrapper>, 'children'>;

export function themeProvider(): ComponentType {
    return  ThemeProvider as ComponentType;
}

export function ThemeProvider({ children }: ThemeProps): ReactElement {
    return (
        <ThemeWrapper theme={equisoftTheme}>
            {children}
        </ThemeWrapper>
    );
}

export function ThemeWrapped(children: ReactElement): ReactElement {
    return (
        <ThemeWrapper theme={equisoftTheme}>
            {children}
        </ThemeWrapper>
    );
}
