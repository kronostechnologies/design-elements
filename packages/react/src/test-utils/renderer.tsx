import { ThemeWrapped } from '@design-elements/test-utils/theme-wrapped';
import { mount, ReactWrapper, render } from 'enzyme';
import React, { Component, ReactElement } from 'react';
import { MemoryRouter } from 'react-router-dom';
import { DeviceContextProvider } from '../components/device-context-provider/device-context-provider';

export function mountWithProviders<C extends Component, P = C['props'], S = C['state']>(
    component: ReactElement<P>,
): ReactWrapper<P, S, C> {
    return mount(component, {
        wrappingComponent: AllProviders,
    });
}

export function renderWithProviders(
    component: ReactElement,
): Cheerio {
    return render(<AllProviders>{component}</AllProviders>);
}

function AllProviders({ children }: { children: ReactElement }): ReactElement {
    return (
        <MemoryRouter>
            <DeviceContextProvider>
                {ThemeWrapped(children)}
            </DeviceContextProvider>
        </MemoryRouter>
    );
}
