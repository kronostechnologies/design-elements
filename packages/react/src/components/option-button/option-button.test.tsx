import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeWrapped } from '../../test-utils/theme-wrapped';
import { OptionButton } from './option-button';

jest.mock('uuid/v4');

describe('Option Button', () => {
    test('Is checked', () => {
        const tree = renderer.create(
            ThemeWrapped(<OptionButton label="option" name="test" value={2} checked />),
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    test('Matches the snapshot', () => {
        const tree = renderer.create(
            ThemeWrapped(<OptionButton label="option" name="test" value={2} />),
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
