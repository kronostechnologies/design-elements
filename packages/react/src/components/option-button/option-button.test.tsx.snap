// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option Button Is checked 1`] = `
.c0 {
  position: absolute;
  z-index: -1;
}

.c0 + label {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FFFFFF;
  border: 1px solid #D9DDE2;
  border-radius: 8px;
  color: #57666E;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 24px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-height: 40px;
}

.c0:checked + label {
  background-color: #0080A5;
  border-color: #0080A5;
  color: #FFFFFF;
}

<div>
  <input
    checked={true}
    className="c0"
    name="test"
    type="radio"
    value={2}
  />
  <label>
    option
  </label>
</div>
`;

exports[`Option Button Matches the snapshot 1`] = `
.c0 {
  position: absolute;
  z-index: -1;
}

.c0 + label {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FFFFFF;
  border: 1px solid #D9DDE2;
  border-radius: 8px;
  color: #57666E;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 24px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-height: 40px;
}

.c0:checked + label {
  background-color: #0080A5;
  border-color: #0080A5;
  color: #FFFFFF;
}

<div>
  <input
    className="c0"
    name="test"
    type="radio"
    value={2}
  />
  <label>
    option
  </label>
</div>
`;
